backend/main.py

```
@@ -0,0 +1,372 @@
+# --- Helper Functions ---
+
+async def get_coordinates(location_name: str):
+    async with httpx.AsyncClient() as client:
+        url = f"https://api.geoapify.com/v1/geocode/search?text={location_name}&apiKey={GEOAPIFY_KEY}"
+        try:
+            response = await client.get(url)
+            data = response.json()
+            if data.get("features"):
+                props = data["features"][0]["properties"]
+                return props["lat"], props["lon"]
+        except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)